pkg/odo/cli/component/deploy.go

```diff
@@ -79,6 +76,13 @@ func (do *DeployOptions) Complete(name string, cmd *cobra.Command, args []string
 func (do *DeployOptions) Validate() (err error) {
 
 	log.Infof("\nValidation")
+
+	envInfo, err := envinfo.NewEnvSpecificInfo(do.componentContext)
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve configuration information")
+	}
+	do.EnvSpecificInfo = envInfo
+
```

Comment:
Any reason why we moved this to the `Validate()` function? I followed the same patterns as the other commands to put it in `Complete()`

Reply:
My misunderstanding of when Complete() and Validate() get called. Backing out this part of change.